Library/Services/Implementation/NfieldSurveyDataService.cs

```diff
@@ -40,14 +74,17 @@ public void InitializeNfieldConnection(INfieldConnectionClient connection)
 
         #endregion
 
-        private INfieldHttpClient Client
+        private static void CheckSurveyId(string surveyId)
         {
-            get { return ConnectionClient.Client; }
+            if (surveyId == null)
+                throw new ArgumentNullException(nameof(surveyId));
+            if (surveyId.Trim().Length == 0)
+                throw new ArgumentException("surveyId cannot be empty");
```

Comment:
if (!Guid.TryParse(surveyId, out var newGuid))
        throw new ArgumentException("surveyId invalid format");

Reply:
we don't check for Ids being a guid anywhere. If we go in that direction it should be something more generic. The surveyId length zero is a check to generate a valid url mainly